docker/lib/dependabot/docker/utils/credentials_finder.rb

```diff
@@ -30,27 +30,43 @@ def credentials_for_registry(registry_hostname)
 
         attr_reader :credentials
 
-        def build_aws_credentials(registry_details)
-          # If credentials have been generated from AWS we can just return them
-          return registry_details if registry_details["username"] == "AWS"
-
-          # If we don't have credentials, we might get them from the proxy
-          return registry_details if registry_details["username"].nil?
+        def get_ecr_client(registry_details, region)
+          if registry_details["use_default_credentials_provider"]
```

Comment:
I think it makes sense to keep this as an option to handle existing uses without credentials.
Could you rename this option to something more specific to avoid any future clashes? Thinking: `use_default_aws_credentials`